ipv8/peerdiscovery/community.py

```diff
@@ -148,12 +177,14 @@ def create_similarity_response(self, identifier, peer):
 
         return self.custom_pack(peer, 2, [auth, dist, payload])
 
-    def create_ping(self):
+    def send_ping(self, peer):
```

Comment:
I think it would be helpful if this `send_ping` would return a `Deferred` that fires if a `pong` message is received. Might be something for a later PR?

Reply:
Nobody should be using this directly, rather the `Peer` object should be used to access the successful ping history.

Reply:
What's the reason why application developers should not use it directly? It is a lightweight method to determine whether a peer is online and connected, which is helpful information in various applications.

Reply:
This is not the correct interface for that. That should be on the `Peer` object level. Otherwise the IPv8 codebase turns into spaghetti.

Reply:
I agree, the `Peer` object is indeed a better location for this.